--- /home/chadh/perl5/lib/perl5/SimpleDB/Client.pm	2010-05-04 21:30:58.000000000 -0230
+++ lib/SimpleDB/Client.pm	2010-12-09 16:29:17.000000000 -0330
@@ -50,7 +50,7 @@

use Moose;
use Digest::SHA qw(hmac_sha256_base64);
-use XML::Bare;
+use XML::Fast;
use LWP::UserAgent;
use HTTP::Request;
use Time::HiRes qw(usleep);
@@ -247,9 +247,10 @@

sub handle_response {
    my ($self, $response) = @_;
-    my $content = eval {XML::Bare::xmlin($response->content)};
+    my $tree = eval { xml2hash($response->content) };
+    my (undef, $content) = each %$tree;  # discard root like XMLin
    # compatibility with SimpleDB::Class
-    if (exists $content->{SelectResult} && $content->{SelectResult} == 1) {
+    if (exists $content->{SelectResult} && ! $content->{SelectResult}) {
        $content->{SelectResult} = {};
    }
    # force an item list into an array
